File: libraries/js/Tweenings/sinusoidal.js
class Sinusoidal {
    static evaluate(tweeningBehaviour, t, b, c, d){
        switch(tweeningBehaviour) {
            case ETweeningBehaviour.IN:
                return Sinusoidal.in(t,b,c,d);
            case ETweeningBehaviour.OUT:
                return Sinusoidal.out(t,b,c,d);
            case ETweeningBehaviour.IN_OUT:
                return Sinusoidal.inOut(t,b,c,d);
            default:
                return null;
        }
    }
    

    static in(t, b, c, d){
        return -c * Math.cos(t / d * (Math.PI / 2)) + c + b;
    }


    static out(t, b, c, d){
        return c * Math.sin(t / d * (Math.PI / 2)) + b;
    }

    static inOut(t, b, c, d){
        return -c / 2 * (Math.cos(Math.PI * t / d) - 1) + b;
    }
}

File: interface/python/Tweenings/sinusoidal.py
from ETweeningBehaviour import ETweeningBehaviour as ETB
import math

class Sinusoidal:
    def evaluate(tweeningBehaviour, t, b, c, d) :
        if tweeningBehaviour == ETB.IN:
            return Sinusoidal.in_sinusoidal(t,b,c,d)
        elif tweeningBehaviour == ETB.OUT:
            return Sinusoidal.out(t,b,c,d)
        elif tweeningBehaviour == ETB.IN_OUT:
            return Sinusoidal.inOut(t,b,c,d)

        return None

    def in_sinusoidal(t, b, c, d):
        return -c * math.cos(t / d * (math.pi / 2)) + c + b
    

    def out(t, b, c, d):
        return c * math.sin(t / d * (math.pi / 2)) + b
    

    def inOut(t, b, c, d):
        return -c / 2 * (math.cos(math.pi * t / d) - 1) + b
    
    